src/sinks/datadog/logs/sink.rs

```diff
@@ -202,16 +202,16 @@ pub fn normalize_as_agent_event(event: &mut Event) {
     }
     if !collisions.is_empty() {
```

Comment:
Unless I'm missing something, I think we could simplify it and approach it in a different way - instead of tracking collisions and moving them into `message.message` we could create a new map - put there all customer attributes AND `message` there and then insert this map under the root-level `message` key. In this case we should have no collisions at all, since we create a new empty object and populate it. The result would be still the same but it would be simpler and better indicate the idea of this change.